Source/Particles/PhotonParticleContainer.cpp

```diff
@@ -168,12 +168,14 @@ PhotonParticleContainer::PushPX (WarpXParIter& pti,
             amrex::ParticleReal Bxp, Byp, Bzp;
             getExternalB(i, Bxp, Byp, Bzp);
 
-            // first gather E and B to the particle positions
-            doGatherShapeN(x, y, z, Exp, Eyp, Ezp, Bxp, Byp, Bzp,
-                           ex_arr, ey_arr, ez_arr, bx_arr, by_arr, bz_arr,
-                           ex_type, ey_type, ez_type, bx_type, by_type, bz_type,
-                           dx_arr, xyzmin_arr, lo, n_rz_azimuthal_modes,
-                           nox, l_lower_order_in_v);
+            if(!do_not_gather){
```

Comment:
wuhu, our new CI works! :)
```
/home/runner/work/WarpX/WarpX/Source/Particles/PhotonParticleContainer.cpp(171): error #3223-D: Implicit capture of 'this' in extended lambda expression
```

Solution: please just assign this to a temporary variable before the `amrex::ParallelFor`.

Reply:
Ah right! I'll do that immediately... great CI!